walrus/methods/input/process_media.py

```diff
@@ -1646,38 +1662,36 @@ def process_one_image_file(self):
 
         """
 
-
-        result = self.read_raw_file()
-        if result is False: return False
+        # Read file if it does not come from a blob
+        if self.input.type != 'from_blob_path':
```

Comment:
let's use the strategy pattern here too 
`strategy = 'default'` or `strategy = 'connection`' feels easier to reason about.

Pablo I'm guessing you would agree but more for others: here the read_raw_file() it's fairly obvious maybe not much difference, but lower down  `shutil.rmtree(self.input.temp_dir)` or not based on the path requires understanding a lot more context

Reply:
Thanks Anthony! Agree that strategy is more readable. However are you picturing renaming the columns to this? Or are you picturing using this columns for the strategy and creating a completely separate function to ingest an image with conections?

Reply:
Seperate functions would be better yes, then `process_one_image_file()` can call either strategy.

Skimming it looks like  moving `self.input.file = File.new(...)` to it's own `create_new_file()` function or similar would handle most of the shared stuff. I'm getting into the details here - up to you - but picturing something like

```
def process_one_image_file: 
  
    .... strategy routing ... then...

    save_connection_data()
OR
    save_default()...

def save_connection_data:
    create_new_file()
   ... connection specific stuff ...

def save_default:
    create_new_file()
   ... default specific stuff ...

```
** names are just for sake of example not suggesting these need to be actual names **

Not sure how much of the connection stuff there is specific to images, but may be some more cross over for other media types, may actually make some parts easier